### PR TITLE
Fix Deadlock With Subscriber Checker

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

In #13226 , we added additional metrics to improve granularity of pubsub metric collection. However one change introduced a deadlock, this PR reverts that change.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

**Review by terencechain — APPROVED** [review]

File: beacon-chain/p2p/pubsub_tracer.go [hunk 1]
```diff
@@ -111,9 +110,9 @@ func (g gossipTracer) setMetricFromRPC(act action, subCtr prometheus.Counter, pu
 		ctrlCtr.WithLabelValues("iwant").Add(float64(len(rpc.Control.Iwant)))
 	}
 	for _, msg := range rpc.Publish {
-		// For incoming messages from pubsub, we validate that the topics are valid
-		// before recording metrics for them.
-		if act == recv && !g.checker.isSubscribedToTopic(*msg.Topic) {
+		// For incoming messages from pubsub, we do not record metrics for them as these values
+		// could be junk.
+		if act == recv {
```
Comment (potuz) [inline beacon-chain/p2p/pubsub_tracer.go:115]: Why this change in logic keeps the validity? I mean how is it that not being subscribed is now valid to report as a counter increase anyway
